graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -129,15 +149,13 @@ BoundQuery schemaQuery(String namespace) {
         .select()
         .column(VERSION_COLUMN_NAME, CONTENTS_COLUMN_NAME)
         .from(namespace, TABLE_NAME)
-        .where(KEY_CONDITION)
+        .where(KEY_COLUMN_NAME, Predicate.EQ, UNIQUE_KEY)
         .build()
         .bind();
   }
 
   /** @return the new version */
-  public SchemaSource insert(String namespace, String newContents) throws Exception {
-
-    ensureTableExists(namespace);
+  public SchemaSource insert(String namespace, String newContents) {
```

Comment:
I removed the `ensureTableExists` call here because `insert` is not the first method we call anymore. If we get here, we can reasonably assume that the table is what we expect. Of course someone could always modify the table at any time, but that is true with any application.